space-plugins/nuxt-base-ui/components/BaseModal.vue

```diff
@@ -0,0 +1,36 @@
+<script setup lang="ts">
+defineProps<{
+	title?: string;
+	message?: string;
+}>();
+
+const modalRef = ref<HTMLDialogElement | null>(null);
+
+const show = () => modalRef.value?.showModal();
+const close = () => modalRef.value?.close();
+
+defineExpose({
+	show,
+	close,
+});
+</script>
+
+<template>
+	<dialog ref="modalRef" class="modal">
+		<form method="dialog" class="modal-box p-0 rounded-lg">
+			<section class="flex flex-col gap-4 p-9 items-center">
+				<slot name="header">
+					<h2 v-if="title" class="font-bold text-xl">
+						{{ title }}
+					</h2>
+				</slot>
+				<slot name="content">
```

Comment:
@BibiSebi, what do you think now? You have the possibility of explicitly adding the header/content or just use title/message props as per default.

Reply:
amazing, love it :O ❤️  

just sharing here conditional slots https://vuejs.org/guide/components/slots.html#conditional-slots in case the user does not provide message nor content slot, then we could hide it in DOM. I did it once for something similar. Again no need to implement it, this is already perfect

Reply:
Oh, in this case, there's no need for that, since the default content is already added via conditional, the slot wouldn't render anything in the DOM ✅ We are covered!